### PR TITLE
Add inline template_content support to the opensearch sink

### LINKED ISSUE #3365: Support inline index templates in OpenSearch sink
**Is your feature request related to a problem? Please describe.**

Data Prepper currently has two built-in index templates - OTel trace and OTel service map. If you wish to bring your own, you currently have two options:

1. Add this as a file on the local file system and tell Data Prepper to use it.
2. Perform a PUT/POST directly on the OpenSearch cluster and not use Data Prepper to add the index template.

For some index templates, it would be nice to have the mapping in the YAML file similar to what we did for Avro schemas in the S3 sink.

**Describe the solution you'd like**

Provide a new field on the `opensearch` sink for providing the index template within the pipeline YAML itself. This new field can be called `template_content` to fit along with the existing `template_file` property.

```
sink:
  - opensearch:
     index: logs-%{yyyy-MM-dd}
     template_content: |
       "template": {
         "aliases": {
             "my_logs": {}
         },
         "settings": {
             "number_of_shards": 2,
             "number_of_replicas": 1
         },
         "mappings": {
           "properties": {
             "timestamp": {
               "type": "date",
               "format": "yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"
             },
             "value": {
               "type": "double"
             }
           }
         }
       }
```

(The above example is using a mapping from the docs: https://opensearch.org/docs/latest/im-plugin/index-templates/)

The `template_content` will be sent as either a composable index template or a v1 template. The `opensearch` sink will use the existing `template_type` property to decide.

**Describe alternatives you've considered (Optional)**

The templates could be provided as a list of templates. However, this is could be problematic:

1. It could become too difficult to read.
2. It would require much more refactoring within the sink. 
3. How would the `opensearch` sink know which index patterns to match on?
4. It does not match the existing `template_file` which uses a single file.
5. Similarly, it would not fit with the existing `ism_policy_file` which uses a single file.

An alternative would be to route data to different `opensearch` sinks and have a template per sink.

Also, we could add support for a list as a future enhancement.

### PR DESCRIPTION
**Pull request by graytaylor0**
### Description
Adds a `template_content` parameter for inline index templates in the OpenSearch sink. This can be used as an alternative to `template_file`
 
### Issues Resolved
Resolves #3365 
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java [hunk 1]
```diff
@@ -177,6 +179,16 @@ public static IndexConfiguration readIndexConfig(final PluginSetting pluginSetti
...
             builder = builder.withTemplateFile(templateFile);
         }
+
+        final String templateContent = pluginSetting.getStringOrDefault(TEMPLATE_CONTENT, null);
+        if (templateContent != null) {
+            builder = builder.withTemplateContent(templateContent);
+        }
+
+        if (templateContent != null && templateFile != null) {
+            LOG.warn("Both template_content and template_file are configured. Only template_content will be used");
```
Comment (dlvenable) [inline data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java:189]: Was there a particular reason you have the content take precedence over the file?

**Review by dlvenable — APPROVED** [review]

**Comment by sharraj** [comment]
Can we call it index template to be inline with standard naming --> https://opensearch.org/docs/latest/im-plugin/index-templates/

File: data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java (hunk 1 above)
Comment (graytaylor0) [reply data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java:189]: Not necessarily but if anything i am figuring the template_content will be more widely used

**Comment by graytaylor0** [comment]
> Can we call it index template to be inline with standard naming --> https://opensearch.org/docs/latest/im-plugin/index-templates/. Like in this API PUT _index_template/ , in pipeline it can be called as index_template

The API for v1 templates is `PUT _template`, which can also be supplied here on template_content

**Comment by dlvenable** [comment]
The existing parameters only use the term "template." Also, if you read the [OpenSearch documentation](https://opensearch.org/docs/latest/im-plugin/index-templates/), you can see that it continues to use the term "template" for generic templates.

This new Data Prepper parameter allows for configuring the template whether it is an "index template" or a "v1 template." Thus, the term "template" appears to fit well here.

**Review by dlvenable — APPROVED** [review]

**Pull request merged**